### PR TITLE
[TorchMhlo] turn on all unary ops tests

### PR DESCRIPTION
**Pull request by fortianyou**

### CODE REVIEW

File: pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/ReduceTensorConversions.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      user->erase();
+    }
+  }
+  return new_type;
+}
+
+template <typename T>
+Value backtraceOperand(Value operand) {
+  auto op = operand.getDefiningOp();
+  if (op && mlir::isa<T>(op)) {
```
Comment (qiuxiafei) [inline pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/ReduceTensorConversions.cpp:44]: nit: What's the difference with this and above lines with `operand.getDefiningOp<T>()`?

File: pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/ReduceTensorConversions.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  SmallVector<Type> new_types;
+  SmallVector<Value> old_args;
+
+  auto n_args = func.getNumArguments();
+  auto& body = func.getBody();
+  for (size_t idx = 0; idx < n_args; ++idx) {
+    Value arg = body.getArgument(idx);
+    Type new_type = arg.getType();
+    old_args.push_back(arg);
+    new_type = removeUser<TensorStaticInfoCastOp>(arg, arg.getType());
```
Comment (qiuxiafei) [inline pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/ReduceTensorConversions.cpp:63]: As documented, this operation may carry shape information. Could it help our shape constraints passes?

File: pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/ReduceTensorConversions.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+#include "torch-mlir/Dialect/TorchConversion/IR/TorchConversionOps.h"
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+using namespace mlir::torch::TorchConversion;
+
+namespace {
+template <typename T>
+Type removeUser(Value arg, Type default_type) {
```
[the remark this answers is not in the document]
Comment (qiuxiafei) [reply pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/ReduceTensorConversions.cpp:28]: Oops .. it moved.

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp [hunk 4]
```diff
@@ -1318,8 +1301,12 @@ class ConvertTorchToMhlo
     INSERT_UNARY_PATTERN(AtenRsqrtOp, mhlo::RsqrtOp)
     INSERT_UNARY_PATTERN(AtenBitwiseNotOp, mhlo::NotOp)
     INSERT_UNARY_PATTERN(AtenCeilOp, mhlo::CeilOp)
+
+    // It's tricky that ConvertOp will use type from the return,
+    // but not from the operand here.
```
[the remark this answers is not in the document]
Comment (fortianyou) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:1306]: Indeed the conversion type should be fetched from the operand.

File: pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/ReduceTensorConversions.cpp (hunk 2 above)
Comment (fortianyou) [reply pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/ReduceTensorConversions.cpp:63]: As discussed offline, this operation doesn't carry shape information. Also, shape constraints passes can get shape information from TensorTypes in the later passes.

File: pytorch_blade/torch_blade_build.py [hunk 5]
```diff
@@ -62,6 +62,10 @@ def __init__(
         self.blade_gemm = check_env_flag(
             "TORCH_BLADE_BUILD_BLADE_GEMM", "OFF"
         )
+        self.skip_disc_cmd_build = check_env_flag(
+            "TORCH_BLADE_SKIP_DISC_CMD_BUILD", "OFF"
```
Comment (Yancey1989) [inline pytorch_blade/torch_blade_build.py:66]: where to use this env var? I can not find usage in this PR.

Comment (fortianyou) [reply pytorch_blade/torch_blade_build.py:66]: Will be used when building. It can be set global.

Comment (Yancey1989) [reply pytorch_blade/torch_blade_build.py:66]: I can not find `TORCH_BLADE_SKIP_DISC_CMD_BUILD` from the codebase except here, for what I mean, we should bring in the flags as little as possible, If we can not enable this flag, always making this flag `OFF` is better.

**Review by Yancey1989 — APPROVED** [review]
LGTM

**Pull request merged**